packages/medusa-payment-adyen/src/subscribers/adyen.js

```diff
@@ -0,0 +1,202 @@
+class AdyenSubscriber {
+  constructor({ cartService, orderService, eventBusService }) {
+    this.cartService_ = cartService
+
+    this.orderService_ = orderService
+
+    this.eventBus_ = eventBusService
+
+    this.eventBus_.subscribe(
+      "adyen.notification_received",
+      async (notification) => this.handleAdyenNotification(notification)
+    )
+  }
+
+  async handleAdyenNotification(notification) {
+    console.log(notification)
+    switch (true) {
```

Comment:
This is weird. 

Do this instead
```js
switch (notification.eventCode) {
  case "AUTHORISATION":
    if (notification.success === "true") {
       handleauth
    } else {
      handlefail
    }
    break
}
```

Reply:
I actually do not agree on this. I think the `switch(true)` makes each case isolated.

Reply:
Fair enough - haven't seen it before but get your point.